project_euler/problem_54/test_poker_hand.py

```diff
@@ -0,0 +1,221 @@
+    RANDOM_TESTS = []
+    for _ in range(N_RANDOM_TESTS):
+        play, oppo = randrange(len(SORTED_HANDS)), randrange(len(SORTED_HANDS))
+        expected = ["Loss", "Tie", "Win"][(play >= oppo) + (play > oppo)]
+        hand, other = SORTED_HANDS[play], SORTED_HANDS[oppo]
+        RANDOM_TESTS.append((hand, other, expected))
+    return RANDOM_TESTS
+
+
+@pytest.mark.parametrize("hand, expected", TEST_FLUSH)
+def test_hand_is_flush(hand, expected):
+    player = PokerHand(hand)
+    assert player._is_flush() == expected
+
+
+@pytest.mark.parametrize("hand, expected", TEST_STRAIGHT)
+def test_hand_is_straight(hand, expected):
+    player = PokerHand(hand)
+    assert player._is_straight() == expected
+
+
+@pytest.mark.parametrize("hand, expected, cards", TEST_FIVE_HIGH_STRAIGHT)
+def test_hand_is_five_high_straight(hand, expected, cards):
+    player = PokerHand(hand)
+    assert player._is_five_high_straight() == expected
+    assert player._cards == cards
+
+
+@pytest.mark.parametrize("hand, expected", TEST_KIND)
+def test_hand_is_same_kind(hand, expected):
+    player = PokerHand(hand)
+    assert player._is_same_kind() == expected
+
+
+@pytest.mark.parametrize("hand, expected", TEST_TYPES)
+def test_hand_values(hand, expected):
+    player = PokerHand(hand)
+    assert player._hand_type == expected
+
+
+@pytest.mark.parametrize("hand, other, expected", TEST_COMPARE)
+def test_compare_simple(hand, other, expected):
+    player, opponent = PokerHand(hand), PokerHand(other)
+    assert player.compare_with(opponent) == expected
+
+
+@pytest.mark.parametrize("hand, other, expected", generate_random_hands())
+def test_compare_random(hand, other, expected):
+    player, opponent = PokerHand(hand), PokerHand(other)
+    assert player.compare_with(opponent) == expected
+
+
+def test_hand_sorted():
+    POKER_HANDS = list(map(PokerHand, SORTED_HANDS))
+    list_copy = POKER_HANDS.copy()
+    shuffle(list_copy)
+    user_sorted = chain(sorted(list_copy))
+    for i, h in enumerate(user_sorted):
+        assert h == POKER_HANDS[i]
+
+
+def test_custom_sort_five_high_straight():
+    """Test that five high straights are compared correctly."""
+    pokerhands = [PokerHand("2D AC 3H 4H 5S"), PokerHand("2S 3H 4H 5S 6C")]
+    pokerhands.sort(reverse=True)
+    assert pokerhands[0].__str__() == "2S 3H 4H 5S 6C"
+
+
+def test_euler_project():
+    """Problem number 54 from Project Euler
+    Testing from poker_hands.txt file."""
+    ans = 0
+    script_dir = os.path.abspath(os.path.dirname(__file__))
+    poker_hands = os.path.join(script_dir, "poker_hands.txt")
+    with open(poker_hands, "r") as file_hand:
+        for line in file_hand:
+            player_hand = line[:14].strip()
+            opponent_hand = line[15:].strip()
+            player, opponent = PokerHand(player_hand), PokerHand(opponent_hand)
+            output = player.compare_with(opponent)
+            if output == "Win":
+                ans += 1
+    assert ans == 376
```

Comment:
Does this match https://github.com/nayuki/Project-Euler-solutions/blob/master/Answers.txt#L54?

Reply:
Yes and all tests mentioned in `test_poker_hand.py` are picked up by Travis and being tested including this one.

Reply:
I think you meant this line: [Problem 054: 376](https://github.com/nayuki/Project-Euler-solutions/blob/82d6bcb97aa2979ab8fa114dc47f6e4627f94aef/Answers.txt#L61)